message_test.go

```diff
@@ -239,6 +239,8 @@ func TestMessagesAppend(t *testing.T) {
 		message, messages := new(Message), new(messages)
 		messages.append(message)
 
+		messages.RLock()
```

Comment:
Tests should also take a lock if you access `messages` internals. Otherwise you get race conditions here, too.